### DISCUSSION TITLE
How to overwrite an asset field

### DISCUSSION THREAD

**Discussion by goellner** [aerni/statamic-advanced-seo#83] [Q&A]
I have the following setup and need to overwrite all the seo images with an image coming from my own Laravel app. 

```php
 // get the layout from statamic
$entry = Entry::find(1);

//get the bundle from elqouent
$bundle = Bundle::where('id', $id)->firstOrFail();

if (!$bundle->published_at) {
    abort(404, "Bundle not found.");
}

$layout = app(EntryResource::class)::make($entry);
$layout->set('seo_og_title', 'this works');
// what do i have to use here to get the asset to be returned? 
// an array, Statamic\Fields\Value or something else?
$layout->set('seo_og_image', ???);
```

My setup is using Statamic as the layout machine, where I have one entry per page and I inject the data from laravel into th API response. I can't use the SEO data from the entry. Since I have some collections like blog, news and so on still in statamic I need to get everything to be returned in the same exact way, so we don't have to write custom logic on the frontend.

**Comment by aerni** [comment]
The `seo_og_image` field is an `Assets` fieldtype and expects a path to an image within the assets container configured in the `advanced-seo.php` config file, e.g. `seo_og_image: seo/image.jpg`.

Your images have to be stored in a Statamic assets container for this to work. Where are your images stored? What kind of value are you working with?
